index.js

```diff
@@ -795,15 +813,13 @@
   //. ['oo!', 'oo?', 'on!', 'on?', 'no!', 'no?', 'nn!', 'nn?']
   //.
   //. > chainRec(
-  //. .   [],
+  //. .   Array,
   //. .   (next, done, x) => Array.isArray(x) ? x.map(next) : [done(x)],
   //. .   [1, [[2, 3], 4], 5]
   //. . )
   //. [1, 2, 3, 4, 5]
   //. ```
-  var chainRec = function(chainRec, f, x) {
-    return ChainRec.getBoundMethod('chainRec', chainRec)(f, x);
-  };
+  var chainRec = ChainRec.staticMethods.chainRec;
 
   //# filter :: (Monad m, Monoid m a) => (a -> Boolean, m a) -> m a
```

Comment:
I think we can make constraints on `m` weeker  ([src](https://wiki.haskell.org/Foldable_and_Traversable#Some_trickier_functions:_concatMap_and_filter)
```hs
filter :: (Applicative f, Foldable f, Monoid (f a)) => (a -> Bool) -> f a -> f a
filter p = foldMap (\a -> if p a then pure a else mempty)
```